CutestTorture/Assets/Scripts/Player/CoinPickup.cs

```diff
@@ -0,0 +1,27 @@
+using UnityEngine;
+
+namespace Player
+{
+    public class CoinPickup : MonoBehaviour
+    {
+        private int _nCoins;
+        public delegate void NotifyCoinsAmountUpdate(int amount);
+        public static event NotifyCoinsAmountUpdate OnCoinPickupEvent;
+    
+        private void Start()
+        {
+            _nCoins = 0;
+        }
+
+        private void OnTriggerEnter(Collider coin)
+        {
+            if (coin.gameObject.name != "Coin(Clone)") // ehh...
```

Comment:
This is not an optimal solution for detecting objects
- It relies on object names (which is error prone)
- It assumes your coins are instantiated (it is an implicit dependency on the level generator. Implicit dependencies are always bad)
- `GameObject.name` create a new string (which is bad for GC)

A simpler way was to use tags, but again, they have some of the mentioned issues. Or check by components, as we did in class.